Clickstream/EventScheduler/Core/EventWarehouser.swift

```diff
@@ -53,7 +53,14 @@ extension DefaultEventWarehouser {
             } else {
                 checkedSelf.batchRegulator.observe(event)
                 checkedSelf.persistence.insert(event)
-                
+                #if EVENT_VISUALIZER_ENABLED
+                /// Update the status of the event to cached
+                /// to check if the delegate is connected, if not no event should be sent to client
+                if let stateViewer = Clickstream._stateViewer {
+                    /// Updating the event state to client to cache based on eventGuid
+                    stateViewer.updateStatus(providedEventGuid: event.guid, state: .cached)
```

Comment:
instead of doing this, like updateStatus, sendEvent to stateviewer; why not listen to changes on the DB itself as a source of truth?

Reply:
I am using the method updateStatus for 4 event states - received, cached, sent to Raccoon and received from Raccoon.
For this reason to get all of the states, I used this function which would map that event guid to its particular state.